docker-compose.yml

```diff
@@ -1,4 +1,27 @@
-version: '3'
+version: "3.8"
+
+x-deploy: &default-deploy
+    replicas: 1
+    restart_policy:
+        condition: on-failure
+
+x-deploy-manager: &default-deploy-manager
+    <<: *default-deploy
+    placement:
+        constraints:
+            - "node.role==manager"
+
+x-deploy-worker: &default-deploy-worker
+    <<: *default-deploy
+    placement:
+        constraints:
+            - "node.role==worker"
+
+x-class-env: &default-class-env
```

Comment:
A more descriptive name to this like "database-env" sounds better?

Reply:
You make a great work, thanks!

Reply:
I was even wondering if we can refactor the database environment to remove duplicacies between `databaseprimary`, `databasesecondary` and `databasearbiter`. However there are some variables which are different so I left those untouched. I feel we can rename `x-class-env` and `default-class-env` to `x-service-database-env` and `default-service-database-env` respectively?

Reply:
Sounds good!